### DISCUSSION TITLE
Include all headers on precompiled distributions.

### DISCUSSION THREAD

**Discussion by cos1110** [raysan5/raylib#3443] [Ideas]
Currently as of using `4.5` (MSVC 64) these headers are distributed by default.
`raylib.h`
`raymath.h`
`rlgl.h`

At some point I needed features located in other headers, so I just went downloaded the master repo zip, and extracted the headers on my library location. Then everything worked fine.

Not so much of a problem, because it took me less than a minute to fix it. But if is there a possible improvement here it would be good to think of it.

If for example, there are only three essential headers provided on the release build, it probably makes the library more beginner-user-friendly and less intimidating. However accessing more advanced headers (eg: `rcamera.h`) means that other experienced users will have to do most often (or even beginners at some point will have to do as well).

So a possible solution is that for release builds, if you want to provide all of the headers, you can give the three basic as they are and the rest inside a subdirectory.
```
include/raylib.h
include/raymath.h
include/rlgl.h

include/extra/config.h
include/extra/rcore.h
...
...

**Comment by Peter0x44** [comment]
The only one which there would be a point of including is rcamera.h
rcore.h is internal to raylib, and so is config.h (including this would be really deceptive, since you need to compile raylib for any of the changes in it to matter)

I think it is a good idea for most users to build raylib - you already have a compiler, use it
both CMake and game-premake can build raylib with your project

**Comment by raysan5** [comment]
Actually I prefer to just include `raylib.h`, it's the only rayñib required header, all the other ones are other single-file hader-only libraries, that can be used independently of raylib or internal raylib headers that shouldn't be exposed.
